src/Components/PropertyInspector/PropertyInspectoryModel.ts

```diff
@@ -344,21 +384,116 @@ abstract class PropertyInspectorModel {
                                                   .mapValue,
                                               path: PropertyInspectorModel.buildPath(
                                                   path,
-                                                  modelProperty.name
+                                                  modelProperty.name,
+                                                  modelProperty.index
                                               ),
                                               propertySourceObject: mapValue,
                                               mapInfo: { key },
                                               isMapChild: true,
                                               forceSet: true,
-                                              schemas
+                                              schemas,
+                                              isArrayItem
                                           }
                                       );
                                   })
                                 : null,
                         ...(schemas && { mapSchemas: schemas })
                     };
                 }
-                case DTDLSchemaType.Array: // TODO support arrays in future
+                case DTDLSchemaType.Array: {
+                    const arrayItems: any[] = PropertyInspectorModel.getPropertyValueOrDefault(
```

Comment:
do we know anything more about these objects at all? Or is `any` really the best we can do here?

Reply:
Could be anything. Not just the DTDL primitives but complex object (maybe not even match the defined DTDL schema).